x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java

```diff
@@ -183,42 +184,24 @@ private XPackSettings() {
     /*
      * Do not allow insecure hashing algorithms to be used for password hashing
      */
-    public static final Setting<String> PASSWORD_HASHING_ALGORITHM = new Setting<>(
-        new Setting.SimpleKey("xpack.security.authc.password_hashing.algorithm"),
+    public static final Setting<String> PASSWORD_HASHING_ALGORITHM = defaultHashingAlgorithmSetting(
+        "xpack.security.authc.password_hashing.algorithm",
         (s) -> {
             if (XPackSettings.FIPS_MODE_ENABLED.get(s)) {
                 return Hasher.PBKDF2_STRETCH.name();
             } else {
                 return Hasher.BCRYPT.name();
             }
-        },
-        Function.identity(),
-        v -> {
-            if (Hasher.getAvailableAlgoStoredHash().contains(v.toLowerCase(Locale.ROOT)) == false) {
-                throw new IllegalArgumentException(
-                    "Invalid algorithm: " + v + ". Valid values for password hashing are " + Hasher.getAvailableAlgoStoredHash().toString()
-                );
-            } else if (v.regionMatches(true, 0, "pbkdf2", 0, "pbkdf2".length())) {
-                try {
-                    SecretKeyFactory.getInstance("PBKDF2withHMACSHA512");
-                } catch (NoSuchAlgorithmException e) {
-                    throw new IllegalArgumentException(
-                        "Support for PBKDF2WithHMACSHA512 must be available in order to use any of the "
-                            + "PBKDF2 algorithms for the [xpack.security.authc.password_hashing.algorithm] setting.",
-                        e
-                    );
-                }
-            }
-        },
-        Property.NodeScope
+        }
     );
 
-    // TODO: This setting of hashing algorithm can share code with the one for password when pbkdf2_stretch is the default for both
```

Comment:
Spotted this TODO while working on https://github.com/elastic/elasticsearch/pull/86079 and figured I might as well!